### ISSUE TITLE
Flameshot cursor is not visible when taking screenshots

### ISSUE THREAD

**Issue by hrs445100058** [flameshot-org/flameshot#2004]
**Flameshot version**
Flameshot v0.10.1
Compiled with Qt 5.12.8

**Describe the bug**
The cursor is not visible when taking a screenshot of Flameshot. I can't accurately click on the function I need. Just like you are blind, I can't see the cursor position.

**System Information**

Linux version 5.13.0-21-generic (buildd@lgw01-amd64-052) (gcc (Ubuntu 11.2.0-7ubuntu2) 11.2.0, GNU ld (GNU Binutils for Ubuntu) 2.37) #21-Ubuntu SMP Tue Oct 19 08:59:28 UTC 2021

ubuntu 20.10    Gnome 40.4.0

**Comment by poilynx** [comment]
desktop environment is `Gnome3` and window manager is `gdm`

**Comment by clomper** [comment]
Same issue here. Cross-hairs disappear when capturing, and no cursor on menu buttons, so can't save, edit, etc without guessing where cursor is. OS is Ubuntu 21.10 DE is Gnome 40.4 and WM is Wayland.

**Comment by mmahmoudian** [comment]
@clomper I'm typing on the phone so I quickly write few points:
1. You didn't need to bump, you literally bumped it 6 days ago by commenting.
2. Wayland is not a Window Manager, it is a Display Server Protocol.
3. You didn't mention your Flameshot version and how you have installed it (from `apt`, `flatpak`, `snap` or `AppImage`)

**Comment by clomper** [comment]
1. Sorry
2. Flameshot 0.9.0+ds1-2 (Debian) | installed, purged, reinstalled with apt-get
3. Gnome 40.4.0

**Comment by mmahmoudian** [comment]
@clomper thanks for the info. The latest stable version is 0.10.2 in which we have done a lot of improvements for Wayland. It is not perfect but we are constantly improving the software. I suggest you to try the new version (for example using Flatpak. You can see instructions and help here:

https://flameshot.org/guide/installation/installation-linux/

But to make it clear, I'm not sure if this cursor issue is resolved in that version, but fir sure it will have major improvements in Wayland.

**Comment by ph-One** [comment]
Been having this issue over the last few months-ish. Sorry I can't be more specific. There is no visible cursor, but hovering over buttons acts as if a cursor is present.

- x11
- Gnome 41.3
- ArcoLinux (had the same issue on Vanilla Arch as well)
- Installation methods that I've tried, all having the same issue:
  - AppImage
  - Git
  - `pacman`

```
Flameshot v11.0.0 (ad1bf28)
Compiled with Qt 5.12.8
linux: 5.16.4-zen1-1-zen
arcolinux: unknown
```

Some output from the AppImage
```
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_init_ssl
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_init_crypto
qt.network.ssl: QSslSocket: cannot resolve ASN1_STRING_get0_data
qt.network.ssl: QSslSocket: cannot resolve EVP_CIPHER_CTX_reset
qt.network.ssl: QSslSocket: cannot resolve RSA_bits
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_new_null
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_push
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_free
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_num
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_pop_free
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_sk_value
qt.network.ssl: QSslSocket: cannot resolve DH_get0_pqg
qt.network.ssl: QSslSocket: cannot resolve SSL_CTX_set_options
qt.network.ssl: QSslSocket: cannot resolve SSL_CTX_set_ciphersuites
qt.network.ssl: QSslSocket: cannot resolve SSL_set_psk_use_session_callback
qt.network.ssl: QSslSocket: cannot resolve SSL_get_client_random
qt.network.ssl: QSslSocket: cannot resolve SSL_SESSION_get_master_key
qt.network.ssl: QSslSocket: cannot resolve SSL_session_reused
qt.network.ssl: QSslSocket: cannot resolve SSL_set_options
qt.network.ssl: QSslSocket: cannot resolve TLS_method
qt.network.ssl: QSslSocket: cannot resolve TLS_client_method
qt.network.ssl: QSslSocket: cannot resolve TLS_server_method
qt.network.ssl: QSslSocket: cannot resolve X509_STORE_CTX_get0_chain
qt.network.ssl: QSslSocket: cannot resolve X509_getm_notBefore
qt.network.ssl: QSslSocket: cannot resolve X509_getm_notAfter
qt.network.ssl: QSslSocket: cannot resolve X509_get_version
qt.network.ssl: QSslSocket: cannot resolve X509_STORE_set_ex_data
qt.network.ssl: QSslSocket: cannot resolve X509_STORE_get_ex_data
qt.network.ssl: QSslSocket: cannot resolve OpenSSL_version_num
qt.network.ssl: QSslSocket: cannot resolve OpenSSL_version
qt.network.ssl: Incompatible version of OpenSSL
Failed to get information about the latest version.  "Host api.github.com not found"
Failed to get information about the latest version.  "Host api.github.com not found"
Failed to get information about the latest version.  "Host api.github.com not found"
```

**Comment by Phosphorus-M** [comment]
> [UPDATE / SOLVED] Not sure exactly how these are related, but I uninstalled `gnome-control-center-system76` and replaced it with `gnome-control-center` (for reasons unrelated to Flameshot) and now I have a cursor. Works with `pacman` (0.10.x) as well as with git version (11.0.0), no problems.


Could be important, I'm using PopOS too with the same problem. 


Update 2025: With the new DE Cosmic everything works well

**Comment by luixal** [comment]
That didn't work out for me, i was using `gnome-control-center` not the system76 version.

Anyhow, I've found something related. When I just start my computer, flameshots works great, I can see the pointer and make selections. After a while (no idea how much, but many time) I hit this bug and I cannot see the pointer. Even if I completely kill flameshot and restart it, **BUT** If take a region screenshot with gnome's default app and the try flameshot again, it works perfectly fine.

No idea why. Maybe the gnome's default app is _"resetting"_ some overlay?

**Comment by All3xJ** [comment]
I have the same issue. The issue is there since a while. I have this bug when I am screen sharing my desktop with Chrome (for example the bug is not happening when I screen share with OBS or with Firefox). I am using GNOME 42.3 Wayland. flameshot-git r1651.4affa92b-1

**Comment by mmahmoudian** [comment]
@luixal  can you reproduce the issue in a clean VM? I cannot and I guess the devs need an environment that can reproduce it. I would be thankful if you can give it a try and see if you have better luck than me, and let us know

**Comment by luixal** [comment]
Hi, I haven't tried. Anyhow, will wait until getting gnome 42 in case it's not flameshot's problem and it gets fixed.

**Comment by lepa22** [comment]
I can confirm that this still exists on Fedora 36 with GNOME 42.

**Comment by yxd0018** [comment]
The missing cursor on Ubuntu 23.04 makes it unusable.

**Comment by jafar-aseel** [comment]
The cursor is not showing up in the Windows 11 too.
System & App Info: 
Flameshot v12.1.0 (96c2c82e)
Compiled with Qt 5.15.2
windows: 11

**Comment by weaverant** [comment]
I am using the latest version Flameshot v13.1.0 (79395fd3) on a freshly updated Windows 11 (25H2 Build 26200.6584).
The mouse pointer is disappearing as soon as I start a capture

**Comment by borgmanJeremy** [comment]
@ElTh0r0  Is this something you have observed in windows?

**Comment by weaverant** [comment]
Yes, the pointer was just on top of the smiley-face, when I took this screenshot

<img width="1208" height="202" alt="Image" src="https://github.com/user-attachments/assets/2a289e1b-ee63-4a78-b020-79b70d2c8cfb" />

**Comment by ElTh0r0** [comment]
> [@ElTh0r0](https://github.com/ElTh0r0) Is this something you have observed in windows?

No, I cannot reproduce it. When I start the capture (by clicking the icon in the taskbar), the mouse cursor is still visible and the cursor changes to a "cross" icon for selecting the screenshot area.

> Yes, the pointer was just on top of the smiley-face, when I took this screenshot

@druf2016 are you using any customized mouse cursor themes?
Just to be sure: It is _not_ about including the mouse cursor in the screenshot itself, is it?

**Comment by weaverant** [comment]
It is about including the mouse cursor in the screenshot. I know, that's no easy thing to do.
But nevermind As long as I know this is the expected behavior, I can live with it.

**Comment by ElTh0r0** [comment]
> It is about including the mouse cursor in the screenshot. I know, that's no easy thing to do. But nevermind As long as I know this is the expected behavior, I can live with it.

This issue here was about complete different topic ("The cursor is not visible when taking a screenshot of Flameshot. I can't accurately click on the function I need. Just like you are blind, I can't see the cursor position.").

Your topic is a feature request; currently it is the expected behaviour, that the cursor is not included on the screenshot. Discussion about such a new feature was started in #604

**Comment by weaverant** [comment]
OK. I am sorry. Must have misread. Please ignore my comments then
